app/src/main/java/org/oppia/android/app/viewmodel/BUILD.bazel

```diff
@@ -0,0 +1,39 @@
+load("@rules_jvm_external//:defs.bzl", "artifact")
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+
+kt_android_library(
+    name = "observable_array_list",
+    srcs = [
+        "ObservableArrayList.kt",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        artifact("androidx.databinding:databinding-adapters"),
+    ],
+)
+
+kt_android_library(
+    name = "observable_view_model",
+    srcs = [
+        "ObservableViewModel.kt",
+    ],
+    visibility = ["//visibility:public"],
+    deps = [
+        artifact("androidx.databinding:databinding-adapters"),
+        artifact("androidx.lifecycle:lifecycle-extensions:2.2.0"),
+    ],
+)
+
+kt_android_library(
+    name = "view_model_provider",
+    srcs = [
+        "ViewModelBridgeFactory.kt",
+        "ViewModelProvider.kt",
```

Comment:
@fsharpasharp Do I have to slip this into two separate targets ?

Reply:
For this instance, it's down to preference since ViewModelBridgeFactory is only used by ViewModelProvider and the dagger generated files. I'd prefer being consistent and only include two or more files in the target if there's a circular dependency. @BenHenning could you weigh in on this?

Reply:
We should make libraries things that are logical groupings of functionality. For most packages, that will be all of the files within that package. Since this is a utility package, I think the current organization is correct (though if I made this I might have actually grouped the two observable classes into one 'observable' package, but I think them being separate is fine).